pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -52,28 +54,30 @@ func (t *Terminator) cordon(ctx context.Context, node *v1.Node) error {
 	return nil
 }
 
-// drain evicts pods from the node and returns true when fully drained
+// drain evicts pods from the node and returns true when non-critical and critical pods are all evicted
 func (t *Terminator) drain(ctx context.Context, node *v1.Node) (bool, error) {
 	// 1. Get pods on node
 	pods, err := t.getPods(ctx, node)
 	if err != nil {
 		return false, fmt.Errorf("listing pods for node %s, %w", node.Name, err)
 	}
-	// 2. Evict pods on node
-	empty := true
-	for _, p := range pods {
-		if !pod.IsOwnedByDaemonSet(p) {
-			empty = false
-			if err := t.coreV1Client.Pods(p.Namespace).Evict(ctx, &v1beta1.Eviction{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: p.Name,
-				},
-			}); err != nil {
-				zap.S().Debugf("Continuing after failing to evict pods from node %s, %s", node.Name, err.Error())
-			}
+	// 2. Separate pods as non-critical and critical
+	// https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
+	nonCritical := []*v1.Pod{}
+	critical := []*v1.Pod{}
+	for _, pod := range pods {
+		if pod.Spec.PriorityClassName != "system-cluster-critical" && pod.Spec.PriorityClassName != "system-node-critical" {
+			nonCritical = append(nonCritical, pod)
+		} else {
+			critical = append(critical, pod)
 		}
 	}
-	return empty, nil
+	// 3. Evict non-critical pods
+	if t.evictPods(ctx, nonCritical) {
```

Comment:
if help() {
  if im() {
    if trapped() {
      if in() {
        if callback() {
          return hell()
        }
      }
    }
  }
}

Reply:
```
if !t.evictPods(ctx, noncritical) {
    return /requeue/ 
}
if !evictPods(ctx, critical) {
   return /requeue/
}
return /done/
```